apps/src/applab/designElements/ThemeDropdown.jsx

```diff
@@ -0,0 +1,84 @@
+// Dropdown for theme selection that shows sample icons of what
+// each theme will look like
+import PropTypes from 'prop-types';
+import React from 'react';
+import {themeOptionsForSelect} from '../constants';
+import Select from 'react-select';
+import 'react-select/dist/react-select.css';
+
+export default class ThemeDropdown extends React.Component {
+  static propTypes = {
+    initialValue: PropTypes.string.isRequired,
+    handleChange: PropTypes.func.isRequired,
+    desc: PropTypes.node
+  };
+
+  state = {
+    selectedValue: this.props.initialValue
+  };
+  handleChange = event => {
+    this.props.handleChange(event.value);
+    this.setState({selectedValue: event.value});
+  };
+
+  // Need to reset theme value if screen switches
+  componentWillReceiveProps(nextProps) {
```

Comment:
Looks like componentWillReceiveProps has been deprecated https://reactjs.org/docs/react-component.html#unsafe_componentwillreceiveprops - perhaps have DesignToolbox control the changeHandler and state. This seems like a component that could be a controlled component.

Reply:
Since the method that is eventually used to update the theme lives in designMode.js, I think the only way to make it fully controlled would be to move everything to designMode.js and then call renderDesignWorkspace when the theme is updated. Is that something we are ok doing (don't know how much strain that puts to re-render the whole workspace)? We would have to do it there because we need to reset the theme when we switch screens.

Reply:
Oh my. Looks like designMode.js was our attempt at re-implementing Redux. 

I have a potential suggestion. First off, doing this "right" is probably outside the scope of this PR. DesignMode needs a good refactor. 

Suggestion: Use `componentDidUpdate` in `DesignToolbox` in order to achieve the same effect that `componentWillReceiveProps` is achieving in `ThemeDropdown`. This is still an anti pattern, but less of one. The [pitfalls described in the reactjs blog](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#anti-pattern-erasing-state-when-props-change) don't seem to apply here. Additionally, this would allow us to make ThemeDropdown a controlled component which may make the eventual refactor a bit easier.

I also noticed that ThemePropertyRow, which you are replacing, used componentWillReceiveProps. I still think we should replace it as part of this change, but I'm up for discussion since you are migrating this part of code rather than adding.

Reply:
Makes sense! I found the "uncontrolled with a key" suggestion [here](https://reactjs.org/blog/2018/06/07/you-probably-dont-need-derived-state.html#recommendation-fully-uncontrolled-component-with-a-key) and found it worked well, and have added a commit with that. Let me know if that works!

Reply:
Oh cool! Good call!